examples/java/src/test/java/org/apache/beam/examples/WordCountIT.java

```diff
@@ -68,15 +66,14 @@
   public void testE2EWordCount() throws Exception {
     PipelineOptionsFactory.register(WordCountITOptions.class);
     WordCountITOptions options = TestPipeline.testingPipelineOptions().as(WordCountITOptions.class);
+
+    IOChannelUtils.registerStandardIOFactories(options);
     options.setOutput(
-        Joiner.on("/")
-            .join(
-                new String[] {
-                  options.getTempRoot(),
-                  String.format("WordCountIT-%tF-%<tH-%<tM-%<tS-%<tL", new Date()),
-                  "output",
-                  "results"
-                }));
+            IOChannelUtils.resolve(
+                    options.getTempRoot(),
+                    IOChannelUtils.resolve(
+                            String.format("WordCountIT-%tF-%<tH-%<tM-%<tS-%<tL", new Date()),
+                            IOChannelUtils.resolve("output", "results"))));
```

Comment:
The first argument to IOChannelUtils.resolve needs to always be the base

Since the default is the FileIOChannelFactory, it will use the file separator which varies based upon OS. So running WordCountIT against Dataflow on Windows would add a "\" between output and results.

You might want to update the method IOChannelUtils.resolve(String base, String path) to be IOChannelUtils.resolve(String base, String ... paths) in a separate PR and fix existing usages of it.